gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/WebHdfsIntegrationTest.java

```diff
@@ -146,12 +146,31 @@ public void testGetDefaultReplication()
 
   /**
    * Validates that we cannot open a non-existent object.
-   * Note: WebHDFS throws IOException (Internal Server Error (error code=500))
+   *
+   * <p>Note: WebHDFS throws IOException (Internal Server Error (error code=500))
    */
-  @Test @Override
-  public void testOpenNonExistent()
-      throws IOException {
-    String bucketName = ghfsHelper.getUniqueBucketName("open-non-existent");
+  @Test
+  @Override
+  public void testOpenNonExistentObject() throws IOException {
+    String bucketName = ghfsHelper.sharedBucketName1;
+    IOException e =
+        assertThrows(
+            IOException.class,
+            () ->
+                ghfsHelper.readTextFile(
+                    bucketName, objectName + "_open-non-existent", 0, 100, true));
+    assertThat(e).hasMessageThat().contains("Internal Server Error (error code=500)");
+  }
+
+  /**
+   * Validates that we cannot open an object in non-existent bucket.
+   *
+   * <p>Note: WebHDFS throws IOException (Internal Server Error (error code=500))
+   */
+  @Test
+  @Override
+  public void testOpenInNonExistentBucket() throws IOException {
```

Comment:
Given we added this case, should we also have a test here like named something like testOpenNonExistentObjectInExistentBucket?

Reply:
Done. Added above.